vault/resource_userpass_user_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAccUserpassUser_basic(t *testing.T) {
+	backend := acctest.RandomWithPrefix("userpass")
+	resName := "vault_userpass_user.user"
+	username := "john_doe"
+	password := "supersecretpassword"
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccUserpassUserCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserpassUserConfig_basic(backend, username, password, []string{"admin", "security"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resName, "id", "auth/"+backend+"/users/"+username),
+					resource.TestCheckResourceAttr(resName, "backend", backend),
+					resource.TestCheckResourceAttr(resName, "username", username),
+					resource.TestCheckResourceAttr(resName, "password", password),
+					resource.TestCheckResourceAttr(resName, "token_policies.#", "2"),
+					resource.TestCheckResourceAttr(resName, "token_policies.0", "admin"),
+					resource.TestCheckResourceAttr(resName, "token_policies.1", "security"),
+				),
+			},
+			{
+				Config: testAccUserpassUserConfig_basic(backend, username, password, []string{"updated_policy"}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resName, "id", "auth/"+backend+"/users/"+username),
+					resource.TestCheckResourceAttr(resName, "backend", backend),
+					resource.TestCheckResourceAttr(resName, "username", username),
+					resource.TestCheckResourceAttr(resName, "password", password),
+					resource.TestCheckResourceAttr(resName, "token_policies.#", "1"),
+					resource.TestCheckResourceAttr(resName, "token_policies.0", "updated_policy"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccUserpassUser_importBasic(t *testing.T) {
```

Comment:
We no longer need separate import tests and can add an import test step to the the test above as mentioned in a comment